oauthenticator/github.py

```diff
@@ -114,6 +114,16 @@ def _userdata_url_default(self):
         """,
     )
 
+    allowed_repositories = Set(
+        config=True,
+        help="""
+        Allow users with read access to specified repositories by specifying
```

Comment:
We also need to think about future maintainability. Could you say a bit about your design decisions to make this conditional on read access as opposed to e.g. write access, making it configurable etc?